triple-store/triple-store-impl-blazegraph/src/main/java/com/powsybl/triplestore/impl/blazegraph/TripleStoreBlazegraph.java

```diff
@@ -228,6 +241,77 @@ public PropertyBags query(String query) {
         }
     }
 
+    @Override
+    public void copyFrom(TripleStore source) {
+        Repository sourceRepository = ((TripleStoreBlazegraph) source).getRepository();
```

Comment:
The restriction on different implementations refers to the triplestore engine libraries (you can not copy from RDF4J to Blazegraph or to Jena, ...) but not on the repository that you are using.
It would be possible to copy an RDF4J triplestore using in-memory repository to an RDF4J triplestore using persistent repository.

Supporting copies between triplestores based on different engine libraries would mean additional work that we think is not (yet) required.

Reply:
we've added the assert on source and check for the object type before casting